components/seldon-request-logger/app/default_logger.py

```diff
@@ -135,11 +135,19 @@ def process_and_update_elastic_doc(
 
             no_items_in_batch = len(new_content_part["instance"])
             index = 0
-            for item in new_content_part["instance"]:
+            elements = None
```

Comment:
this was just iterating through the instances in case of batch... now we need to iterate through instances and find corresponding 'elements' for each but only if there are elements (might not be if there are no names and no schema)